src/virtualenv/interpreters/discovery/py_spec.py

```diff
@@ -0,0 +1,115 @@
+"""A Python specification is an abstract requirement definition of a interpreter"""
+from __future__ import absolute_import, unicode_literals
+
+import os
+import re
+import sys
+from collections import OrderedDict
+
+PATTERN = re.compile(r"^(?P<impl>[a-zA-Z]+)?(?P<version>[0-9.]+)?(?:-(?P<arch>32|64))?$")
+IS_WIN = sys.platform == "win32"
+
+
+class PythonSpec(object):
+    """Contains specification about a Python Interpreter"""
+
+    def __init__(self, str_spec, implementation, major, minor, patch, architecture, path):
+        self.str_spec = str_spec
+        self.implementation = implementation
+        self.major = major
+        self.minor = minor
+        self.patch = patch
+        self.architecture = architecture
+        self.path = path
+
+    @classmethod
+    def from_string_spec(cls, string_spec):
+        impl, major, minor, patch, arch, path = None, None, None, None, None, None
```

Comment:
```suggestion
        impl = major = minor = patch = arch = path = None
```

slightly less counting necessary when adding / removing a variable ;)